frontend/src/components/pages/acls/UserCreate.tsx

```diff
@@ -415,6 +416,59 @@ export const RoleSelector = observer((p: { state: string[] }) => {
   );
 });
 
+// use instead of RoleSelector whn not using mobx
+export const StateRoleSelector = ({ roles, setRoles }: { roles: string[]; setRoles: (roles: string[]) => void }) => {
+  const [searchValue, setSearchValue] = useState('');
+  const {
+    data: { roles: allRoles },
+  } = useListRolesQuery();
+  const availableRoles = (allRoles ?? []).filter((r) => !roles.includes(r.name)).map((r) => ({ value: r.name }));
+
+  return (
+    <Flex direction="column" gap={4}>
+      <Box w="280px">
+        <Select<string>
+          isMulti={false}
+          options={availableRoles}
+          inputValue={searchValue}
+          onInputChange={setSearchValue}
+          placeholder="Select roles..."
+          noOptionsMessage={() => 'No roles found'}
+          // TODO: Selecting an entry triggers onChange properly.
+          //       But there is no way to prevent the component from showing no value as intended
+          //       Seems to be a bug with the component.
+          //       On 'undefined' it should handle selection on its own (this works properly)
+          //       On 'null' the component should NOT show any selection after a selection has been made (does not work!)
+          //       The override doesn't work either (isOptionSelected={()=>false})
```

Comment:
Thanks for the writeup. I wouldn't change legacy component library at this point, just focus on getting us to a more modern setup.

Reply:
To be honest most of this componenet is a copy paste from RoleSelector I just changed to useState rather than mobx